iceoryx_hoofs/source/cxx/helplets.cpp

```diff
@@ -21,27 +21,33 @@ namespace iox
 {
 namespace cxx
 {
+// AXIVION DISABLE STYLE AutosarC++19_03-A5.2.4 : allow reinterpret_cast for the whole file
 void* alignedAlloc(const uint64_t alignment, const uint64_t size) noexcept
 {
     // -1 == since the max alignment addition is alignment - 1 otherwise the
     // memory is already aligned and we have to do nothing
     // low-level memory management
     // NOLINTNEXTLINE(cppcoreguidelines-owning-memory, cppcoreguidelines-no-malloc, hicpp-use-auto)
-    uint64_t memory = reinterpret_cast<uint64_t>(malloc(size + alignment + sizeof(void*) - 1));
+    // AXIVION Next Line AutosarC++19_03-M5.2.9 : Pointer conversion is safe here.
+    // AXIVION Next Line AutosarC++19_03-A18.5.1 : Dynamic memory management function is required.
+    const uint64_t memory = reinterpret_cast<uint64_t>(malloc((size + alignment + sizeof(void*)) - 1));
     if (memory == 0)
     {
         return nullptr;
     }
-    uint64_t alignedMemory = align(memory + sizeof(void*), alignment);
-    assert(alignedMemory >= memory + 1);
+    
+    const uint64_t alignedMemory = align(memory + sizeof(void*), alignment);
+    assert(alignedMemory >= (memory + 1));
     // low-level memory management
     // NOLINTNEXTLINE(performance-no-int-to-ptr, cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    // AXIVION Next Line AutosarC++19_03-M5.2.8 : Cast required for conversion and has no side-effects.
     reinterpret_cast<void**>(alignedMemory)[-1] = reinterpret_cast<void*>(memory);
 
     // NOLINTNEXTLINE(performance-no-int-to-ptr) low-level memory management
     return reinterpret_cast<void*>(alignedMemory);
 }
 
+// AXIVION Construct AutosarC++19_03-M7.1.2 : idiomatic for deallocate
 void alignedFree(void* const memory) noexcept
```

Comment:
Could you also rename `memory` to `alignedMemory` to match the naming of `alignedAlloc`.